spec/lib/granite/action/performer_spec.rb

```diff
@@ -13,6 +13,16 @@ def self.batch(count)
     end
   end
 
+  describe '#ctx' do
+    specify { expect(Action.new.ctx).to eq({}) }
+    specify { expect(Action.using(key: :value).new.ctx).to eq(key: :value) }
+    specify { expect(Action.as(performer1).new.ctx).to eq(performer: performer1) }
+
+    specify 'proxy works for deeper initialization' do
+      expect(Action.using(key: :value).batch(2).map(&:ctx)).to eq([{key: :value}, {key: :value}])
+    end
+  end
+
```

Comment:
Please add test of thread safety. Changing context in another thread does not affect the context of current thread.